src/book.js

```diff
@@ -341,20 +345,18 @@ export class Book {
         (this.source_rotation == 'none' ? '' : `_${this.source_rotation}`);
       this.filename = `${origFileName}${rotationMetaInfo}`;
       resultPDF = aggregatePdf0;
-    } else if (this.format == 'a9_3_3_4') {
-      resultPDF = await this.buildSheets(this.filename, this.book.a9_3_3_4_builder());
-    } else if (this.format == 'a10_6_10s') {
-      resultPDF = await this.buildSheets(this.filename, this.book.a10_6_10s_builder());
-    } else if (this.format == 'a_4_8s') {
-      resultPDF = await this.buildSheets(this.filename, this.book.a_4_8s_builder());
-    } else if (this.format == 'a_3_6s') {
-      resultPDF = await this.buildSheets(this.filename, this.book.a_3_6s_builder());
-    } else if (this.format == 'A7_2_16s') {
-      resultPDF = await this.buildSheets(this.filename, this.book.a7_2_16s_builder());
-    } else if (this.format == '1_3rd') {
-      resultPDF = await this.buildSheets(this.filename, this.book.page_1_3rd_builder());
-    } else if (this.format == '8_zine') {
-      resultPDF = await this.buildSheets(this.filename, this.book.page_8_zine_builder());
+    } else if (
+      [
+        'a9_3_3_4',
+        'a10_6_10s',
+        'a_4_8s',
+        'a_3_6s',
+        'a7_2_16s',
+        'page_1_3rd',
+        'page_8_zine',
+      ].includes(this.format)
+    ) {
+      resultPDF = await this.buildSheets(this.filename, this.book[`${this.format}_builder`]());
```

Comment:
I don't love this (too close to `eval` for my liking) but willing to accept it temporarily - ideally the WackyImposition class would have a generic builder that takes a config name and returns the correct builder, probably by doing a lookup on an object (with the config name being the keys and builder references being the values)

Reply:
yeah, I agree, but I think it's a step in a better direction.

Reply:
changed to use a map of names to functions so this is less delicate

Reply:
Actually nevermind I hate that too, changed to switch cases.